src/talos/HPOFlagging.py

```diff
@@ -0,0 +1,249 @@
+"""
+Takes the result date from ValidateMOI
+For each reportable result, identify if the variant is a strong phenotype match for the family
+
+1. Write 2 results - the result set annotated with Phenotype-match flags
+2. The annotated result set, reduced to only phenotype matches
+
+Include something here about where to find the inputs ...
+
+gene_to_phtnotype: Download: https://hpo.jax.org/app/data/annotations#:~:text=download-,GENES,-TO%20PHENOTYPE"
+phenio.db: Download: https://data.monarchinitiative.org/monarch-kg/latest/phenio.db.gz
+"""
+
+from argparse import ArgumentParser
+from collections import defaultdict
+
+from semsimian import Semsimian
+
+from talos.config import config_retrieve
+from talos.models import ParticipantResults, ResultData
+from talos.static_values import get_granular_date
+from talos.utils import phenotype_label_history, read_json_from_path
+
+_SEMSIM_CLIENT: Semsimian | None = None
+
+
+def get_sem_client(phenio_db: str | None = None) -> Semsimian:
+    """
+    create or retrieve a Semsimian client
+
+    Args:
+        phenio_db (str | None): needs to be present for the first call
+
+    Returns:
+        Semsimian instance
+    """
+    global _SEMSIM_CLIENT
+    if _SEMSIM_CLIENT is None:
+        _SEMSIM_CLIENT = Semsimian(spo=None, predicates=['rdfs:subClassOf'], resource_path=phenio_db)
+    return _SEMSIM_CLIENT
+
+
+def parse_genes_to_hpo(g2p_file: str, ensg2symbol: str, genes: set[str]) -> dict[str, set[str]]:
+    """
+    Parse genes to phenotype file from Jax.
+    Returns a dict of gene_symbol -> set of HPO ids
+
+    Args:
+        g2p_file ():
+        ensg2symbol (): the gene symbols to Ensembl gene IDs for genes relevant to this analysis
+        genes (set[str]): all genes in this report
+
+    Returns:
+        dict[str, set[str]]: ENSG -> set of HPO ids
+    """
+
+    # lookup of {symbol: ENSG}
+    ensg_map: dict[str, str] = read_json_from_path(ensg2symbol)
+
+    gene_to_phenotype: dict[str, set[str]] = defaultdict(set)
+    with open(g2p_file, encoding='utf-8') as f:
+        for line in f:
+            ncbi_gene_id, gene_symbol, hpo_id, hpo_name, frequency, disease_id = line.split('\t')
+            if gene_symbol in ensg_map and ensg_map[gene_symbol] in genes:
+                gene_to_phenotype.setdefault(ensg_map[gene_symbol], set()).add(hpo_id)
+    return gene_to_phenotype
+
+
+def find_genes_in_these_results(result_object: ResultData) -> set[str]:
+    """
+    identify all the genes we care about finding phenotype matches for
+
+    Args:
+        result_object (ResultData):
+
+    Returns:
+        set[str], all ensembl gene IDs in
+    """
+    ensgs: set[str] = set()
+    for participant in result_object.results.values():
+        for variant in participant.variants:
+            # get the gene ID
+            ensgs.add(variant.var_data.info['gene_id'])
+
+            # for structural variants, add all the LOF'd genes
```

Comment:
We should leave some TODO comment here noting that if we ever extend our SV interp beyond LOFs then this will need revising.